tests/test_download_puller.py

```diff
@@ -0,0 +1,112 @@
+import os
+import pytest
+import shutil
+import nbgitpuller.plugin_helper as ph
+import importlib
+import aiohttp
+from aioresponses import aioresponses
+google_nb = importlib.import_module("nbgitpuller.plugins.nbgitpuller-googledrive.googledrive_puller")
+
+test_files_dir = os.getcwd() + "/tests/test_files"
+archive_base = "/tmp/test_files"
+repo_parent_dir = "/tmp/fake/"
+temp_download_repo = "/tmp/download/"
+temp_archive_download = "/tmp/archive_download/"
+provider = "dropbox_test"
+url = "http://test/this/repo"
+CACHED_ORIGIN_NON_GIT_REPO = ".nbgitpuller/targets/"
+origin_repo = f"{repo_parent_dir}{CACHED_ORIGIN_NON_GIT_REPO}{provider}/{url}/"
+
+repo_zip = 'file://' + archive_base + ".zip"
+repo_tgz = 'file://' + archive_base + ".tar.gz"
+
+
+@pytest.fixture
+async def test_configuration():
+    shutil.make_archive(archive_base, 'zip', test_files_dir)
+    shutil.make_archive(archive_base, 'gztar', test_files_dir)
+    os.makedirs(temp_archive_download, exist_ok=True)
+    os.makedirs(repo_parent_dir, exist_ok=True)
+    os.makedirs(temp_download_repo, exist_ok=True)
+    yield "test finishing"
+    os.remove(archive_base + ".zip")
+    os.remove(archive_base + ".tar.gz")
+    if os.path.isfile(temp_archive_download + "downloaded.zip"):
+        os.remove(temp_archive_download + "downloaded.zip")
+    shutil.rmtree(repo_parent_dir)
+    shutil.rmtree(temp_download_repo)
+    shutil.rmtree(temp_archive_download)
+
+
+def test_extract_file_extension():
+    url = "https://fake.com/master/materials-sp20-external.tgz"
```

Comment:
Is fake.com has made me think its a website that provides actual dummy responses. But I understand now by inspecting it, that it isn't.

```suggestion
    url = "https://example.local/master/materials-sp20-external.tgz"
```

Reply:
`example.com`/`example.org` are official example domains https://www.iana.org/domains/reserved